svf/include/SVFIR/SVFVariables.h

```diff
@@ -344,11 +367,15 @@ class ObjVar: public SVFVar
  */
 class ArgValVar: public ValVar
 {
+    friend class GraphDBClient;
 
 private:
     const FunObjVar* cgNode;
     u32_t argNo;
 
+protected:
+    ArgValVar(NodeID i, const SVFType* type, u32_t argNo, PNODEK ty = ArgValNode) : ValVar(i, type, ty), argNo(argNo) {}
```

Comment:
We need to revisit many of the protected constructors you added, in fact you could reuse the existing constructors by passing null for some arguments.